requests/models.py

```diff
@@ -907,7 +911,13 @@ def json(self, **kwargs):
                     # and the server didn't bother to tell us what codec *was*
                     # used.
                     pass
-        return complexjson.loads(self.text, **kwargs)
+
+        try:
+            return json.loads(self.text, **kwargs)
+        except json.JSONDecodeError as e:
+            # Catch JSON-related errors and raise as requests.JSONDecodeError
+            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError
+            raise JSONDecodeError(e.msg, e.doc, e.pos)
```

Comment:
We should probably preserve the original exception instance in this as well and proxy attributes down to it. That way folks can find the original exception. I'm not certain `simplejson` and `json` have the same exception attributes with the same meaning

Reply:
I could catch a `ValueError` and that will get caught for both `simplejson` and `json` exceptions. Sound good?

Reply:
@sigmavirus24  Actually, at least in the latest versions of both `json` and `simplejson`, they both have `msg`, `doc`, and `pos` attributes. I'll add an extra `except` for `simplejson`.

Reply:
Both will get caught as `json.JSONDecodeError` since either are retrieved from `compat.py` as `json`.

Reply:
I'm not sure I'm following how this will work in the Python 2 `json` case. Where is `ValueError` being handled?

Reply:
https://github.com/psf/requests/pull/5856/checks?check_run_id=3031103164 Yeah, this is broken. We need to be able to handle this a different way.

Reply:
I see your last comment was made outside of this thread, @steveberdy. To clarify, while `json.JSONDecodeError` does inherit from `ValueError`, it's not going to work in this case. The exception doesn't exist in Python 2, so we fall over with an attribute error when handling the exception. You'll need to take another approach to handle this.

Reply:
Hi @nateprewitt. I'm sorry if this has been frustrating -- I've just found this comment. I'll be working on this in `compat.py`. Do you know of any good ways to test a Python 2 environment when I have Python 3 installed?